web/srv/api_handlers.go

```diff
@@ -155,3 +155,75 @@ func (h *handler) handleApiPods(w http.ResponseWriter, req *http.Request, p http
 
 	renderJsonPb(w, pods)
 }
+
+func (h *handler) handleApiStat(w http.ResponseWriter, req *http.Request, p httprouter.Params) {
+	timeWindow := req.FormValue("window")
+	resourceName := req.FormValue("resource_name")
+	resourceType := req.FormValue("resource_type")
+	namespace := req.FormValue("namespace")
+	outToResourceName := req.FormValue("out_to_resource_name")
+	outToResourceType := req.FormValue("out_to_resource_type")
+	outToResourceNamespace := req.FormValue("out_to_resource_namespace")
+	outFromResourceName := req.FormValue("out_from_resource_name")
+	outFromResourceType := req.FormValue("out_from_resource_type")
+	outFromResourceNamespace := req.FormValue("out_from_resource_namespace")
+
+	window := defaultMetricTimeWindow
+	if timeWindow != "" {
+		var requestedWindow pb.TimeWindow
+		requestedWindow, err := util.GetWindow(timeWindow)
+		if err != nil {
+			renderJsonError(w, err, http.StatusInternalServerError)
+			return
+		}
+		window = requestedWindow
+	}
+
+	statRequest := &pb.StatSummaryRequest{
+		Selector: &pb.ResourceSelection{
+			Resource: &pb.Resource{
+				Namespace: namespace,
+				Name:      resourceName,
+				Type:      resourceType,
+			},
+		},
+		TimeWindow: window,
+	}
+
+	if outToResourceName != "" || outToResourceType != "" || outToResourceNamespace != "" {
+		if outToResourceNamespace == "" {
+			outToResourceNamespace = namespace
+		}
+
+		outToResource := pb.StatSummaryRequest_OutToResource{
+			OutToResource: &pb.Resource{
+				Namespace: outToResourceNamespace,
+				Type:      outToResourceType,
+				Name:      outToResourceName,
+			},
+		}
+		statRequest.Outbound = &outToResource
+	}
+
+	if outFromResourceName != "" || outFromResourceType != "" || outFromResourceNamespace != "" {
+		if outFromResourceNamespace == "" {
+			outFromResourceNamespace = namespace
+		}
+
+		outFromResource := pb.StatSummaryRequest_OutFromResource{
+			OutFromResource: &pb.Resource{
+				Namespace: outFromResourceNamespace,
+				Type:      outFromResourceType,
+				Name:      outFromResourceName,
+			},
+		}
+		statRequest.Outbound = &outFromResource
```

Comment:
similar logic exists in `cli/cmd/stat_summary.go`'s `buildStatSummaryRequest`. any way to share?

Reply:
added `util.BuildStatSummaryRequest()`